### PR TITLE
Backport quarkus CLI update support

### PR DESCRIPTION
**Pull request by mocenas**
### Summary

Backport of changes for quarkus CLI update tests. From PRs:
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1241
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1239
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1236
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1235
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1233
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1232
- https://github.com/quarkus-qe/quarkus-test-framework/pull/1188

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Release (follows conventions described in the [RELEASE.md](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/RELEASE.md))
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [ ] Example scenarios has been updated / added
- [ ] Methods and classes used in PR scenarios are meaningful
- [ ] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: examples/quarkus-cli/src/test/resources/existingSourcesApp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+
+    <properties>
+        <compiler-plugin.version>3.13.0</compiler-plugin.version>
+        <maven.compiler.release>17</maven.compiler.release>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+        <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+        <quarkus.platform.version>3.12.3</quarkus.platform.version>
```
Comment (michalvavrik) [inline examples/quarkus-cli/src/test/resources/existingSourcesApp/pom.xml:15]: Quarkus 3.12.3 feels wrong in FW branch dedicated to 3.8, but alright.

**Comment by michalvavrik** [comment]
CLI failures look related.

File: examples/quarkus-cli/src/test/resources/existingSourcesApp/pom.xml (hunk 1 above)
Comment (mocenas) [reply examples/quarkus-cli/src/test/resources/existingSourcesApp/pom.xml]: You're right it would be better to have 3.8 version here. I've changed it.

**Comment by mocenas** [comment]
> CLI failures look related.

I investigated that. In the end I've disabled the test and opened an upstream issue, as this really seems like an problem in quakrus CLI in version 3.8.*. https://github.com/quarkusio/quarkus/issues/42567
If steps like this doesn't work (even with upstream versions) then there is something wrong:
```
quarkus create app app --stream=3.2
cd app
quarkus update --stream=3.8
```

**Comment by rsvoboda** [comment]
Why one bulk commit and not picks of the original commits (using `git cherry-pick -x __HASH__`)?

btw, `-x` is important as you will have reference to the original commit and thus the PR for better context.

**Comment by michalvavrik** [comment]
> I investigated that. In the end I've disabled the test and opened an upstream issue, as this really seems like an problem in quakrus CLI in version 3.8.*. [quarkusio/quarkus#42567](https://github.com/quarkusio/quarkus/issues/42567)
> If steps like this doesn't work (even with upstream versions) then there is something wrong:

makes sense

**Comment by mocenas** [comment]
> Why one bulk commit and not picks of the original commits (using `git cherry-pick -x __HASH__`)?
> 
> btw, `-x` is important as you will have reference to the original commit and thus the PR for better context.

To be frank, I tried to do that way, but git cherry-pick was trolling me so hard, that I decided to go this way instead.

**Comment by mocenas** [comment]
I've also disabled the shouldUpdateApplication test which is failing for same reason - https://github.com/quarkusio/quarkus/issues/42567

**Review by michalvavrik — APPROVED** [review]

**Pull request merged**